ktor-client/ktor-client-tests/posix/test/io/ktor/client/tests/HttpClientFreezingTest.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.tests
+
+import io.ktor.client.*
+import io.ktor.client.request.*
+import io.ktor.client.tests.utils.*
+import io.ktor.utils.io.*
+import kotlin.test.*
+
+class HttpClientFreezingTest : ClientLoader() {
+
+    @Test
+    fun testRequestWithFrozenClient() = clientTests {
+        test { client ->
+            client.makeShared()
```

Comment:
Will the clients that use `HttpClient` also call `makeShared` or is it just for testing?

Reply:
It's just for testing cases when user share client in the different coroutine dispatcher or create a client on the top-level scope